src/components/AddBook.js

```diff
@@ -1,14 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
 import '../styles/AddBook.css';
+import Button from './Button';
 
 function AddBook() {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const booksArr = useSelector((state) => state.book.books);
+  const dispatch = useDispatch();
+  const clickHandler = (e) => {
+    const id = booksArr.length + 1;
+    dispatch(addBook({
+      itemId: `item${id + 1}`,
+      title,
+      author,
+    }));
```

Comment:
- Well done. 👍 I like the way you increased item id. Though it seems easy, most of the students missed it.
- Keep it up and happy coding. 😄